src/app/components/Pagination.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div class="flex items-center items-center justify-center py-6">
+    <div
+      v-if="currentPage !== 1"
+      @click="changePage(currentPage - 1)"
+      class="arrow arrow-left py-1 px-2 mx-1 bg-indigo-300 hover:bg-indigo-600 with-transition rounded-full cursor-pointer"
+    ></div>
+
+    <div v-for="(page, idx) in pagesArray" :key="idx">
+      <p
+        v-if="typeof page === 'number'"
+        @click="changePage(page)"
+        :class="paginationClassObject(page)"
+      >{{ page }}</p>
+      <p v-else-if="page === 'dots' && shoultRenderDots(idx)" class="py-1 px-2 mx-1">...</p>
+    </div>
+
+    <div
+      v-if="currentPage !== pagesCount"
+      @click="changePage(currentPage + 1)"
+      class="arrow arrow-right py-1 px-2 mx-1 bg-indigo-300 hover:bg-indigo-600 with-transition rounded-full cursor-pointer"
+    ></div>
+  </div>
+</template>
+
+<script>
+import { mapActions } from 'vuex';
+
+export default {
+  props: {
+    limit: Number,
+    offset: Number,
+    total: Number
+  },
+
+  computed: {
+    pagesCount() {
+      return Math.floor(this.total / this.limit) + 1;
+    },
+
+    currentPage() {
+      return Number(this.$route.query.page) || 1;
+    },
+
+    pagesArray() {
+      if (this.pagesCount <= 10) {
+        let arr = [];
+        for (let i = 1; i <= this.pagesCount; i++) {
+          arr.push(i);
+        }
+        return arr;
+      } else if (this.currentPage <= 3) {
+        return [
+          1,
+          2,
+          3,
+          4,
+          'dots',
+          this.pagesCount - 2,
+          this.pagesCount - 1,
+          this.pagesCount
+        ];
+      } else if (this.currentPage >= this.pagesCount - 2) {
+        return [
+          1,
+          'dots',
+          this.pagesCount - 3,
+          this.pagesCount - 2,
+          this.pagesCount - 1,
+          this.pagesCount
+        ];
+      } else {
+        return [
+          1,
+          'dots',
+          this.currentPage - 2,
+          this.currentPage - 1,
+          this.currentPage,
+          this.currentPage + 1,
+          this.currentPage + 2,
+          'dots',
+          this.pagesCount
+        ];
+      }
+    }
+  },
+
+  methods: {
+    ...mapActions('search', ['fetchItems']),
+
+    changePage(pageNumber) {
+      if (pageNumber === this.currentPage) return;
+
+      const title = this.$route.query.title;
+
+      this.$router.push({ query: { title, page: pageNumber } });
+
+      this.fetchItems({
```

Comment:
It would be better to not mix logic of pagination and fetching the data.
It should only display a pagination view based on props and call some callback with the clicked page. That's it